tedana/decomposition/eigendecomp.py

```diff
@@ -423,20 +429,28 @@ def tedica(n_components, dd, fixed_seed):
 
     if fixed_seed == -1:
         fixed_seed = np.random.randint(low=1, high=1000)
-    rand_state = np.random.RandomState(seed=fixed_seed)
-    ica = FastICA(n_components=n_components, algorithm='parallel',
-                  fun='logcosh', max_iter=5000, random_state=rand_state)
 
-    with warnings.catch_warnings(record=True) as w:
-        # Cause all warnings to always be triggered.
-        warnings.simplefilter('always')
+    for i_attempt in range(maxrestart):
+        ica = FastICA(n_components=n_components, algorithm='parallel',
+                      fun='logcosh', max_iter=maxit, random_state=fixed_seed)
 
-        ica.fit(dd)
+        with warnings.catch_warnings(record=True) as w:
+            # Cause all warnings to always be triggered.
+            warnings.simplefilter('always')
 
-        w = list(filter(lambda i: issubclass(i.category, UserWarning), w))
-        if len(w):
-            LGR.warning('ICA failed to converge')
+            ica.fit(dd)
+
+            w = list(filter(lambda i: issubclass(i.category, UserWarning), w))
+            if len(w):
+                LGR.warning('ICA attempt {0} failed to converge after {1} '
+                            'iterations'.format(i_attempt + 1, ica.n_iter_))
+                fixed_seed += 1
+                LGR.warning('Random seed updated to {0}'.format(fixed_seed))
+            else:
+                LGR.info('ICA attempt {0} converged in {1} '
+                         'iterations'.format(i_attempt + 1, ica.n_iter_))
+                break
 
     mmix = ica.mixing_
     mmix = stats.zscore(mmix, axis=0)
-    return mmix, fixed_seed
+    return mmix
```

Comment:
Is there a reason to not return the `fixed_seed` value, now ? I think it'd still be nice to have to keep provenance, but maybe there's a reason to remove it that I'm missing....

Reply:
I think we must have originally used it in `io.writeresults` for something (filenames maybe?), but that function no longer uses the seed so it's not actually used anywhere. Plus, the log file will now have that info, along with number of iterations required for convergence, for those who might want it.

Reply:
We did use it, I think to go in the summary file. But if we're ok that it's not just in the log file, that sounds fine by me ! :+1: